types/field_type.go

```diff
@@ -261,8 +261,8 @@ func DefaultTypeForValue(value interface{}, tp *FieldType, char string, collate
 		tp.Flag |= mysql.UnsignedFlag
 		SetBinChsClnFlag(tp)
 	case BinaryLiteral:
-		tp.Tp = mysql.TypeBit
-		tp.Flen = len(x) * 8
+		tp.Tp = mysql.TypeVarString
+		tp.Flen = len(x)
```

Comment:
What does this change mean?

Reply:
set @a = 0x3135;
    prepare stmt 'select * from t1 where col_varchar = ?';
    execute stmt using @a;

The quesction mark should be TypeVarString, otherwise when convert 0x3135(TypeBIt) to int, the result if 12597, Equal expr will return false.